controllers/manila_controller.go

```diff
@@ -676,8 +681,7 @@ func (r *ManilaReconciler) reconcileNormal(ctx context.Context, instance *manila
 		instance.Status.Conditions.Set(c)
 	}
```

Comment:
Yes, mostly for `Mirror` the condition, `ReadyCount` is assigned and refreshed but I would have an explicit:

```golang
  
if !shareObsGen {
    instance.Status.Conditions.Set(condition.FalseCondition(
	manilav1beta1.ManilaShareReadyCondition,
	condition.ErrorReason,
	condition.SeverityWarning,
	manilav1beta1.ManilaShareReadyErrorMessage,
	err.Error()))
}
```
that also makes easy in the code to see that the `observedGeneration` influences the `DeploymentReady` condition

Reply:
That could work. (I would use Unknown instead of False though as it is not a terminal error but something still in progress)

Reply:
False is in case there's an error while getting the sub resources, but Unknown is ok if `observedGeneration` doesn't match and the result is actually False.